### ISSUE TITLE
[$2000] Chat - Deleted image URL crashes the app on mac and windows (works fine on mobile)

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22261]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app and login with user A (device 1)
2. Open any group chat
3. Login with any other user that is in the above group in another device (device 2)
4. Send any image attachment from device 2 in group chat
5. From device 1, open the attachment in preview, copy the URL and send it in the group
6. From device 2, delete the attachment sent in step 4
7. From device 1, click on the URL and observe that app crashes instead of showing 'Hmm its not here' page like it does on android chrome

## Expected Result:
App should display 'Hmm its not here' page when we open link of delete image like it does on android chrome

## Actual Result:
App crashes when we open link of delete image

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.36-4
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/ba3b4fad-0ce0-4461-a168-3f86b3e55d04

https://github.com/Expensify/App/assets/93399543/46d534ad-cb0d-424c-8932-b9b647e364a0

**Expensify/Expensify Issue URL:**
**Issue reported by:**  @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688488679783719

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0113703c85388c5be7</li>
        <li>Upwork Job ID: 1678516819849125888</li>
        <li>Last Price Increase: 2023-08-02</li>
    </ul>
</details>

**Comment by ahmedGaber93** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Chat - Deleted image URL crashes the app on mac and windows
### What is the root cause of that problem?
when attachment not found we throw js error not handle the error with UI
https://github.com/Expensify/App/blob/f9a8529f18696b2584d57c960933cefdf7b2e6af/src/components/AttachmentCarousel/index.js#L187-L190
### What changes do you think we should make in order to solve the problem?

```javascript

we need to handle when page = -1 and show error screen

if (page === -1) {
      this.setState({showNotFound: true});
      return;
}

//wrap content with
<FullPageNotFoundView
     shouldShow={this.state.showNotFound}
     ...
     />
     
     
</FullPageNotFoundView>     

```
### What alternative solutions did you explore? (Optional)

**Comment by twisterdotcom** [comment]
@mananjadhav - Looks like we have a few proposals here that might work.

**Comment by mananjadhav** [comment]
Checking those today.

**Comment by mananjadhav** [comment]
@twisterdotcom Thanks for the bump. I reviewed the proposals, and both have the correct RCA. I just need to verify the pros and cons of both. I'll have that confirmed by tomorrow.

**Comment by mananjadhav** [comment]
Thanks for the patience here. Apologies folks, it takes time when we have similar proposals with edited history. I think @ahmedGaber93's proposal covered most expects from the beginning and hence we should be using their proposal.

🎀 👀 🎀 C+ reviewed.

**Comment by marcochavezf** [comment]
Thanks for the review @mananjadhav, assigning @ahmedGaber93 🚀

**Comment by ahmedGaber93** [comment]
@mananjadhav [PR](https://github.com/Expensify/App/pull/23143) is now ready for review.

**Comment by mananjadhav** [comment]
We had the PR finished, but the component was refactored to functional component and had conflicts. We've resolved this but need sometime to review again and test.

**Comment by mananjadhav** [comment]
@marcochavezf We've got some conflicts due to another issue and I've raised a comment to clarify the change. https://github.com/Expensify/App/issues/21334#issuecomment-1656859609

**Comment by marcochavezf** [comment]
Seems the other PR caused a [regression](https://github.com/Expensify/App/issues/21334#issuecomment-1658472307). Waiting for a resolution there.

**Comment by marcochavezf** [comment]
Seems that the other issue was agreed on [dismiss the modal](https://github.com/Expensify/App/issues/21334#issuecomment-1660049928), this is a bummer because seems the implementation would change in this PR, thoughts? @mananjadhav @ahmedGaber93

**Comment by ahmedGaber93** [comment]
@marcochavezf @mananjadhav 
from this [comment](https://expensify.slack.com/archives/C01GTK53T8Q/p1688964378953149?thread_ts=1688646398.596839&cid=C01GTK53T8Q) they agree to dismiss modal only for their issue and leave us to handle our issue.
> I think those are separate. The issue you linked is as vistiting any other resource you dont ahve access to or does not exist. I think we can continue here with dynamically closing the modal and we can leave that other issue to be handled separately

We can do the both NotFoundPage for deeplinks, and dismiss modal for live update.
The other issue use not accurate condition for live update page === -1.
https://github.com/Expensify/App/blob/024d210bc0b5a0454b62a72de60057af67cdd045/src/components/AttachmentCarousel/createInitialState.js#L53-L55

I suggest remove it from here it use it inside this [use effect](https://github.com/Expensify/App/blob/024d210bc0b5a0454b62a72de60057af67cdd045/src/components/AttachmentCarousel/index.js#L51-L56)
```javascript
    useEffect(() => {
        const initialState = createInitialState(props);
        // if file is found in prevState attachments and not found in newState attachments.
        // this mean the file is deleted in live update so dismiss modal here.
        if(_.find(attachments, (a) => a.source === props.source) && !_.find(initialState.attachments, (a) => a.source === props.source)) {
            Navigation.dismissModal();
        }
        ...
    }, [props.reportActions]);
```

What do you think about this fix? @mananjadhav @marcochavezf

**Comment by mananjadhav** [comment]
Yeah I think we can do that. My concern is that `_.find` for a lot of attachments could be expensive? I am not aware maximum number of attachments in the chat. Are you aware @ahmedGaber93 ?

**Comment by ahmedGaber93** [comment]
@mananjadhav 
> I am not aware maximum number of attachments in the chat. Are you aware @ahmedGaber93 ?

I am also not aware of that, but we already use `_.findIndex `through attachments here.

https://github.com/Expensify/App/blob/024d210bc0b5a0454b62a72de60057af67cdd045/src/components/AttachmentCarousel/createInitialState.js#L52

and we can simplify it to
```javascript
// replace this !_.find(initialState.attachments, (a) => a.source === props.source
// with initialState.page === -1
// it calculate in createInitialState no need to calculate it again
if(_.find(attachments, (a) => a.source === props.source) && initialState.page === -1)) {


// it found in oldState   
// _.find(attachments, (a) => a.source === props.source) 

// it not found in newState   
// initialState.page === -1
```

**Comment by twisterdotcom** [comment]
I am happy to increase to 2000 for the increased work. Can we retain @ahmedGaber93 as the assignee for this?

**Comment by mananjadhav** [comment]
Thanks @twisterdotcom. Yes we can retain @ahmedGaber93.

**Comment by mananjadhav** [comment]
I was away for the weekend, and I'll be reviewing this tomorrow.

**Comment by mananjadhav** [comment]
We're again blocked by conflicts causing additional changes. @ahmedGaber93 is working through it.

**Comment by twisterdotcom** [comment]
Apologies but I am finally going on Parental Leave. I am reassigning to another member of the team.

**Comment by twisterdotcom** [comment]
Apologies but I am finally going on Parental Leave. I am reassigning to another member of the team.

**Comment by mananjadhav** [comment]
This was deployed on production 2 days back but the title wasn't updated.

**Comment by mananjadhav** [comment]
@mallenexpensify Can you post the payout summary? This is in production for 7 days, but the title wasn't updated.

**Comment by mananjadhav** [comment]
The `Attachment not found` was intentionally added, and we didn't consider this a bug until we had this issue. Moreover based on multiple iterations, we moved through `dismissModal`, etc. but all of them were expected behaviors. Hence there is no regression PR.

But I do think we should add regression test as it involves App crash. The test steps from the [PR](https://github.com/Expensify/App/pull/23143#issue-1810868025) are good for the regression steps here. wdyt @marcochavezf @mallenexpensify ?


@mallenexpensify quick bump on the previous comment for the payout summary.

**Comment by mallenexpensify** [comment]
@dhanashree-sawant can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~01e88042a91e290cf6

For issue payment, I see `days elapsed: 22` which would mean a 50% penalty.  Is there any reason this shouldn't be applied @mananjadhav and @marcochavezf ?

TestRail GH created https://github.com/Expensify/Expensify/issues/313988

@twisterdotcom  I'm off this week, can you please keep 👀 on this then I'll snag it back on Monday?  Thx

**Comment by mananjadhav** [comment]
@mallenexpensify the scope of the issue was changed multiple times. Hence we increased the bounty to 2000. I don't think there would be any penalty here, and also no timeline bonus.

**Comment by mallenexpensify** [comment]
Thanks @mananjadhav 

Issue reporter: @dhanashree-sawant paid $250 via Upwork
Contributor: @ahmedGaber93 paid $2000 via Upwork
Contributor+: @mananjadhav is due $2000 via NewDot. 

Closing.. I think we got everything.. right?

**Comment by mananjadhav** [comment]
@mallenexpensify I think this needs to be opened until @JmillsExpensify approves it on NewDot.

**Comment by mallenexpensify** [comment]
It shouldn't need to, but I'll reopen, assign to Mills and let him close again.  Per the [internal StackOverflow post](https://stackoverflowteams.com/c/expensify/questions/16949/16950#16950)
> If you are assigned an issue that they are the C+ on, you can close the issue as soon as the Contributor and bug reporter and regression steps have been completed.

**Comment by mananjadhav** [comment]
Okay in previous issues I've seen issues kept open. Waiting for @JmillsExpensify to process this one though.

**Comment by JmillsExpensify** [comment]
$2,000 payment approved via NewDot based on BZ summary.
